google/cloud/bigquery/v2/minimal/internal/job_rest_stub_test.cc

```diff
@@ -82,71 +95,129 @@ TEST(BigQueryJobStubTest, GetJobSuccess) {
   EXPECT_THAT(result->job.status.state, Eq("DONE"));
 }
 
-TEST(BigQueryJobStubTest, ProjectIdEmpty) {
+TEST(BigQueryJobStubTest, GetJobRestClientError) {
+  // Get() fails.
   auto mock_rest_client = std::make_unique<MockRestClient>();
-  GetJobRequest job_request("", "j123");
+  EXPECT_CALL(*mock_rest_client, Get(An<rest::RestRequest const&>()))
+      .WillOnce(
+          Return(rest::AsStatus(HttpStatusCode::kInternalServerError, "")));
+
+  GetJobRequest job_request("p123", "j123");
   Options opts;
   opts.set<EndpointOption>("bigquery.googleapis.com");
   rest_internal::RestContext context;
   DefaultBigQueryJobRestStub rest_stub(std::move(mock_rest_client), opts);
   auto result = rest_stub.GetJob(context, std::move(job_request));
-  EXPECT_THAT(
-      result,
-      StatusIs(StatusCode::kInvalidArgument,
-               HasSubstr("Invalid GetJobRequest: Project Id is empty")));
+  EXPECT_FALSE(result.ok());
+  EXPECT_THAT(result.status().code(), Eq(StatusCode::kUnavailable));
 }
 
-TEST(BigQueryJobStubTest, JobIdEmpty) {
+TEST(BigQueryJobStubTest, GetJobRestResponseError) {
+  // Invalid Rest response.
+  auto mock_payload = std::make_unique<MockHttpPayload>();
+  auto mock_response = std::make_unique<MockRestResponse>();
+  EXPECT_CALL(*mock_response, StatusCode)
+      .WillRepeatedly(Return(HttpStatusCode::kBadRequest));
+  EXPECT_CALL(std::move(*mock_response), ExtractPayload)
+      .WillOnce(Return(std::move(mock_payload)));
+
+  // Get() is successful.
   auto mock_rest_client = std::make_unique<MockRestClient>();
-  GetJobRequest job_request("p123", "");
+  EXPECT_CALL(*mock_rest_client, Get(An<rest::RestRequest const&>()))
+      .WillOnce(Return(ByMove(
+          std::unique_ptr<rest::RestResponse>(std::move(mock_response)))));
```

Comment:
What value are we expecting for `*mock_response` here, given that we moved it above?